src/main/java/org/cryptomator/cloudaccess/vaultformat8/DirectoryIdCache.java

```diff
@@ -32,8 +33,19 @@
 		}
 	}
 
+	public void evict(CloudPath cleartextPath) {
+		cache.remove(cleartextPath);
+	}
+
+	public CompletionStage<Void> evictFolderIncludingChilds(CloudPath cleartextPath) {
+		return CompletableFuture.supplyAsync(() -> {
+			var subPaths = cache.keySet().stream().filter(path -> path.startsWith(cleartextPath)).collect(Collectors.toSet());
+			subPaths.forEach(cache::remove);
+			return null;
+		});
```

Comment:
I don't think this qualifies for async execution. Spawning a background task is probably more expensive than iterating over a few thousand entries (worst case!).

Good old Iterator-over-EntrySet-loop should be fine and is easier to debug, too (no race conditions).